extensions/amp-fx-collection/0.1/providers/amp-fx-presets-utils.js

```diff
@@ -52,3 +53,123 @@ export function resolvePercentageToNumber(val) {
     return precentageStrippedVal / 100;
   }
 }
+
+export function installStyles(ampdoc, element, fxType, flyInDistance) {
+  const style = computedStyle(ampdoc.win, element);
+  switch (fxType) {
+    case 'parallax':
+      return {
+        'will-change': 'transform',
+      };
+    case 'fade-in':
+      return {
+        'will-change': 'opacity',
+        'opacity': 0,
+      };
+    case 'fade-in-scroll':
+      return {
+        'will-change': 'opacity',
+        'opacity': 0,
+      };
+    case 'fly-in-bottom':
+      return {
+        'will-change': 'transform',
+        'top': 'calc(' + style.top + ' + ' + flyInDistance + 'vh)',
+        'visibility': 'visible',
+      };
+    case 'fly-in-top':
+      return {
+        'will-change': 'transform',
+        'top': 'calc(' + style.top + ' - ' + flyInDistance + 'vh)',
+        'visibility': 'visible',
+      };
+    case 'fly-in-left':
+      return {
+        'will-change': 'transform',
+      };
+    case 'fly-in-right':
+      return {
+        'will-change': 'transform',
+      };
+    default:
+      return {
+        'visibility': 'visible',
+      };
+  }
+}
+
+export function defaultDurationValues(ampdoc, fxType) {
+  switch (fxType) {
+    case 'fade-in':
+      return '1000ms';
+    case 'fly-in-bottom':
+    case 'fly-in-top':
+    case 'fly-in-left':
+    case 'fly-in-right':
+      const screenWidth = ampdoc.win.screen.width;
+      if (screenWidth <= 480) { // mobile devices
```

Comment:
How about `mapRange(Math.min(1000, screenWidth), 480, 1000, 400, 600) + 'ms'` instead of breakpoints?

Reply:
I honestly believe this is the wrong way to go, but if breakpoints should be transparent to devs (which I also believe they shouldn't be), then:

1. it would be nice if you hoisted the breakpoints as an object at the top of the file so that they're clear when first reading the file.

2. Use `const {width} = Services.viewportForDoc(win).getSize();` to prevent layout thrash.

Reply:
The breakpoints are exposed to the devs as:
X is the value on Mobile, Y on tablets and Z on desktop. Why do you think that shouldn't be the case?

1. Exposed the numbers as consts at the start of the file. 
2. Done.

Reply:
Because screen size is not a proxy for device type, nor should it be. Here we're making some assumptions about layout already by defining hardcoded percentages, and I don't think it's the best idea to fragment that further by splitting things in such a way.

Again, not a dealbreaker for me, but I just think ranging is simpler and thus better. I'm imagining the set of presets growing in the future, and having all of them contain specific breakpoints and mappings in different parts. It will be difficult to maintain and will also be rather large in size.

Reply:
Hi @alanorozco, as per offline discussion, I polled a few web devs to see what is the expected behaviour there. The points brought up are as follows:
* To keep the speed at which the element moves constant across all devices, the default should be a continuous function. (3 people brought this up)
* 1 person liked the idea of different discrete default values as long as you could then set these values separately. E.g. be able to set data-mobile-duration='300ms', data-tablet-duration='400ms'. Since we don't support that, the defaults need to be better - hence use continuous function. 
* Folks who know about responsive layout designing (hopefully all good developers) will already be expecting this and will find this concept easy to understand. (2 people brought this up)

Given that, I have added new commits to address your concerns. Thank you for being patient while I polled people to see what the developers would expect and find easier to use. :) PTAL?

Code change is here: https://github.com/ampproject/amphtml/pull/14782/commits/35e5114b9e8f2d82ab72165840d87c0eb5482810
Edits to markdown are here: https://github.com/ampproject/amphtml/pull/14782/commits/7dc85cd309a1c7e5165bb2f2a01204db89b5c754